core/main/src/broker/endpoint_broker.rs

```diff
@@ -420,36 +420,44 @@ impl BrokerOutputForwarder {
                         {
                             let request_id = rpc_request.ctx.call_id;
                             v.data.id = Some(request_id);
-                            if let Some(result) = v.data.result.clone() {
-                                if is_event {
-                                    if let Some(filter) = broker_request
-                                        .rule
-                                        .transform
-                                        .get_filter(super::rules_engine::RuleTransformType::Event)
-                                    {
-                                        if let Ok(r) = jq_compile(
-                                            result,
-                                            &filter,
-                                            format!("{}_event", rpc_request.ctx.method),
-                                        ) {
-                                            v.data.result = Some(r);
-                                        }
-                                    }
-                                } else if let Some(filter) = broker_request
+
+                            let jv: Value = if let Some(result) = v.data.clone().result {
+                                result
+                            } else if let Some(error) = v.data.clone().error {
```

Comment:
Please add the error handling in ThunderBroker::update_response

fn update_response(response: &JsonRpcApiResponse) -> JsonRpcApiResponse {
        let mut new_response = response.clone();
        if response.params.is_some() {
            new_response.result = response.params.clone();
        } else if response.error.is_some() {
            new_response.error = response.error.clone();
         }
        new_response
    }

Reply:
"new_response.error = response.error.clone();"

error is already in response

// response=JsonRpcApiResponse { jsonrpc: "2.0", id: Some(1), result: None, **error:** Some(Object {"code": Number(43)}), method: None, params: None }

Not sure you meant "new_response.**result** = response.error.clone();"
Fist I did without this, then with this.

More changes needed to return an error back to the caller in endpoint_broker.